connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryOffsetBackingStore.java

```diff
@@ -52,11 +52,15 @@ public void configure(WorkerConfig config) {
 
     @Override
     public synchronized void start() {
+        executor = Executors.newSingleThreadExecutor();
     }
 
     @Override
     public synchronized void stop() {
-        // Nothing to do since this doesn't maintain any outstanding connections/data
+        if (executor != null) {
+            executor.shutdown();
+            executor = null;
```

Comment:
`shutdown()` only triggers shutdown, but it does not wait for the tasks to be completed. How about adding a `awaitTermination` ?